### ISSUE TITLE
 all "host" failure domains: []. osd is down in failure domain: "". active node drains: false. pg health: "cluster is not fully clean. PGs: [{StateName:unknown Count:60}]"

### ISSUE THREAD

**Issue by vyom-soft** [rook/rook#12008]
Hi,
I created a rook-ceph cluster via helm chart on new **k8s cluster v1.26.3**, new disk.
helm install -f ./values.yaml rook-ceph rook-release/rook-ceph --version 1.10.7  --namespace rook-ceph
image: quay.io/ceph/ceph:v17.2.3

I am seeing that no osd are created. host" failure domains: []. osd is down in failure domain.
Pls. advise.

```
Disk /dev/sdb: 1 TiB, 1099511627776 bytes, 2147483648 sectors
Units: sectors of 1 * 512 = 512 bytes
Sector size (logical/physical): 512 bytes / 512 bytes
I/O size (minimum/optimal): 512 bytes / 512 bytes 
```

```
lsblk -l
NAME          FSTYPE      LABEL UUID                                   MOUNTPOINT
sda
|-sda1        vfat              5F7D-C125                              /boot/efi
|-sda2        xfs               d2cc7d78-e0ec-486f-bebb-5bb8dcf62139   /boot
`-sda3        LVM2_member       clzjyk-SRnA-LnZf-f2GV-id1P-zL5t-ApwKmz
  |-rhel-root xfs               14005b17-abdd-46d7-838f-ab551433787c   /
  `-rhel-swap swap              9d0ee4d4-679d-4ac2-9e34-35223aee8f8d
sdb
```
```
lsblk -o KNAME,TYPE,SIZE,MODEL
KNAME TYPE  SIZE MODEL
sda   disk   32G Virtual disk
sda1  part  600M
sda2  part    1G
sda3  part 30.4G
sdb   disk    1T Virtual disk
dm-0  lvm  27.2G
dm-1  lvm   3.2G

```
This is the storage
```
  storage: # cluster level storage configuration and selection
    useAllNodes: false
    useAllDevices: false
    deviceFilter: sd[b-z]
    config:
    nodes:
      - name: "node3"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
      - name: "node4"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
      - name: "node5"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
      - name: "node6"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
      - name: "node7"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
      - name: "node8"
        devices: # specific devices to use for storage can be specified for each node
          - name: "sdb"
```
I also zapped the disk
 sgdisk --zap-all /dev/sdb
 dd if=/dev/zero of=/dev/sdb bs=1M count=100 oflag=direct,dsync
 partprobe /dev/sdb
```
Portion of operator log

c000e15600 b:0xc000e15640 c:0xc000e15680]}
2023-03-30 11:35:57.597821 I | ceph-spec: detecting the ceph image version for image quay.io/ceph/ceph:v17.2.3...
2023-03-30 11:35:57.602429 D | op-k8sutil: ConfigMap ceph-object-controller-detect-version is already deleted
2023-03-30 11:35:58.511812 D | cephclient: checking any images/snapshots present in pool "ceph-blockpool"
2023-03-30 11:35:58.511844 D | exec: Running command: rbd pool stats ceph-blockpool --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
2023-03-30 11:35:59.192981 D | CmdReporter: job ceph-object-controller-detect-version has returned results
2023-03-30 11:35:59.210053 I | ceph-spec: detected ceph image version: "17.2.3-0 quincy"
2023-03-30 11:35:59.210093 D | exec: Running command: ceph versions --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
2023-03-30 11:35:59.238135 D | ceph-spec: object "ceph-object-controller-detect-version" did not match on delete
2023-03-30 11:35:59.238185 D | ceph-spec: object "ceph-object-controller-detect-version" did not match on delete
2023-03-30 11:35:59.238191 D | ceph-spec: object "ceph-object-controller-detect-version" did not match on delete
2023-03-30 11:35:59.238211 D | ceph-spec: object "ceph-object-controller-detect-version" did not match on delete
2023-03-30 11:35:59.497785 D | clusterdisruption-controller: pg health check timeout is not set in the cluster. waiting for PGs to get active+clean
2023-03-30 11:35:59.497820 I | clusterdisruption-controller: all "host" failure domains: []. osd is down in failure domain: "". active node drains: false. pg health: "cluster is not fully clean. PGs: [{StateName:unknown Count:60}]"
2023-03-30 11:35:59.499512 D | op-k8sutil: kubernetes version fetched 1.26.3
2023-03-30 11:35:59.505492 D | op-k8sutil: kubernetes version fetched 1.26.3
2023-03-30 11:35:59.505537 I | clusterdisruption-controller: reconciling osd pdb reconciler as the allowed disruptions in default pdb is 0
2023-03-30 11:35:59.505585 D | clusterdisruption-controller: reconciling "rook-ceph/"
2023-03-30 11:35:59.506589 D | op-k8sutil: kubernetes version fetched 1.26.3
2023-03-30 11:35:59.507522 D | op-k8sutil: kubernetes version fetched 1.26.3
2023-03-30 11:35:59.507617 D | exec: Running command: ceph status --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json
2023-03-30 11:35:59.710589 D | cephclient: {"mon":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":3},"mgr":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":2},"osd":{},"mds":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":2},"overall":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":7}}
2023-03-30 11:35:59.710620 D | cephclient: {"mon":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":3},"mgr":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":2},"osd":{},"mds":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":2},"overall":{"ceph version 17.2.3 (dff484dfc9e19a9819f375586300b3b79d80034d) quincy (stable)":7}}
2023-03-30 11:35:59.710738 D | ceph-cluster-controller: updating ceph cluster "rook-ceph" status and condition to &{Health:{Status:HEALTH_WARN Checks:map[MDS_SLOW_METADATA_IO:{Severity:HEALTH_WARN Summary:{Message:1 MDSs report slow metadata IOs}} PG_AVAILABILITY:{Severity:HEALTH_WARN Summary:{Message:Reduced data availability: 60 pgs inactive}} TOO_FEW_OSDS:{Severity:HEALTH_WARN Summary:{Message:OSD count 0 < osd_pool_default_size 3}}]} FSID:3b2f2eb4-0a8f-49af-8a70-a712c4672952 ElectionEpoch:12 Quorum:[0 1 2] QuorumNames:[a b c] MonMap:{Epoch:3 FSID: CreatedTime: ModifiedTime: Mons:[]} OsdMap:{OsdMap:{Epoch:0 NumOsd:0 NumUpOsd:0 NumInOsd:0 Full:false NearFull:false NumRemappedPgs:0}} PgMap:{PgsByState:[{StateName:unknown Count:60}] Version:0 NumPgs:60 DataBytes:0 UsedBytes:0 AvailableBytes:0 TotalBytes:0 ReadBps:0 WriteBps:0 ReadOps:0 WriteOps:0 RecoveryBps:0 RecoveryObjectsPerSec:0 RecoveryKeysPerSec:0 CacheFlushBps:0 CacheEvictBps:0 CachePromoteBps:0} MgrMap:{Epoch:0 ActiveGID:0 ActiveName: ActiveAddr: Available:true Standbys:[]} Fsmap:{Epoch:63 ID:1 Up:1 In:1 Max:1 ByRank:[{FilesystemID:1 Rank:0 Name:ceph-filesystem-b Status:up:creating Gid:14494}] UpStandby:1}}, True, ClusterCreated, Cluster created successfully
2023-03-30 11:35:59.710753 D | ceph-spec: CephCluster "rook-ceph" status: "Ready". "Cluster created successfully"
2023-03-30 11:35:59.816366 D | ceph-cluster-controller: checking for stuck pods on not ready nodes
2023-03-30 11:35:59.816493 D | ceph-spec: found 1 ceph clusters in namespace "rook-ceph"
2023-03-30 11:35:59.816503 D | ceph-cluster-controller: update event on CephCluster CR
```

**Comment by sp98** [comment]
@vyom-soft Can you share the osd prepare pod logs. Thanks.

**Comment by vyom-soft** [comment]
> @vyom-soft Can you share the osd prepare pod logs. Thanks.

I will do so. Meanwhile, I have destroyed the rook-ceph cluster again.

**Comment by sp98** [comment]
Hi. Not able to open the `rook-ceph-operator.txt` file that you have attached in the above comment.  Can you please provide the rook operator pod logs. That would help in debugging.

**Comment by sp98** [comment]
> I have destroyed the rook-ceph cluster again. Zapped it. & Fresh deployed again. Unfortunately there is no osd prepare pod seen ?. Pls. can can you tell where can I see the osd prepare pod. Earlier also I did not see osd prepare pod.

It is possible that you are looking at the cluster at a later point of time and failing OSD prepare pods have been deleted by then.  If you are not seeing the OSD prepare pods, then I would suggest to restart the rook operator pod (just delete the pod and it will restarted). Once the operator is restarted, then you can see the OSD prepare pod.

**Comment by vyom-soft** [comment]
> Hi. Not able to open the `rook-ceph-operator.txt` file that you have attached in the above comment. Can you please provide the rook operator pod logs. That would help in debugging.

Hi, Thank you for you time. The fresh operator log. Please see the same.
[rook-ceph-operator-log.txt](https://github.com/rook/rook/files/11118628/rook-ceph-operator-log.txt)

**Comment by vyom-soft** [comment]
> > I have destroyed the rook-ceph cluster again. Zapped it. & Fresh deployed again. Unfortunately there is no osd prepare pod seen ?. Pls. can can you tell where can I see the osd prepare pod. Earlier also I did not see osd prepare pod.
> 
> It is possible that you are looking at the cluster at a later point of time and failing OSD prepare pods have been deleted by then. If you are not seeing the OSD prepare pods, then I would suggest to restart the rook operator pod (just delete the pod and it will restarted). Once the operator is restarted, then you can see the OSD prepare pod.

Thank you. I restarted the operator by deleting the operator pod. Unfortunately there is no osd prepare pod seen.

```
➜  ~ k get pods -n rook-ceph
NAME                                               READY   STATUS     RESTARTS          AGE
csi-cephfsplugin-29jq6                             2/2     Running    0                 10h
csi-cephfsplugin-2kfs7                             2/2     Running    0                 10h
csi-cephfsplugin-2pn4f                             2/2     Running    0                 10h
csi-cephfsplugin-6tp7q                             2/2     Running    0                 10h
csi-cephfsplugin-bpjbz                             2/2     Running    0                 10h
csi-cephfsplugin-jc8t6                             2/2     Running    0                 10h
csi-cephfsplugin-provisioner-b4bb5f74d-2xh4w       5/5     Running    0                 10h
csi-cephfsplugin-provisioner-b4bb5f74d-knwdc       5/5     Running    0                 10h
csi-cephfsplugin-s677j                             2/2     Running    0                 10h
csi-cephfsplugin-zplwq                             2/2     Running    0                 10h
csi-rbdplugin-9ch5g                                2/2     Running    0                 10h
csi-rbdplugin-dwsww                                2/2     Running    0                 10h
csi-rbdplugin-frvp8                                2/2     Running    0                 10h
csi-rbdplugin-gwwnq                                2/2     Running    0                 10h
csi-rbdplugin-n2qct                                2/2     Running    0                 10h
csi-rbdplugin-p48dm                                2/2     Running    0                 10h
csi-rbdplugin-provisioner-578fc9c599-g5kpk         5/5     Running    0                 10h
csi-rbdplugin-provisioner-578fc9c599-wfvct         5/5     Running    0                 10h
csi-rbdplugin-ptvbq                                2/2     Running    0                 10h
csi-rbdplugin-tw8jj                                2/2     Running    0                 10h
rook-ceph-detect-version-9jnt9                     0/1     Init:0/1   0                 1s
rook-ceph-mds-ceph-filesystem-a-69484b7ff5-cbbgw   2/2     Running    0                 9h
rook-ceph-mds-ceph-filesystem-b-7cf8cf45b4-thzz5   2/2     Running    0                 9h
rook-ceph-mgr-a-d54c4f85f-9mh7z                    3/3     Running    0                 9h
rook-ceph-mgr-b-86db4899d-rmls6                    3/3     Running    0                 9h
rook-ceph-mon-a-6bb8998d7d-znsw9                   2/2     Running    0                 10h
rook-ceph-mon-b-6c78679c8-4b4tw                    2/2     Running    0                 10h
rook-ceph-mon-c-55598c547d-hmnvt                   2/2     Running    0                 10h
rook-ceph-operator-677d7d9c4d-6d78c                1/1     Running    0                 13m
rook-ceph-rgw-ceph-objectstore-a-84bbd6745-5lqdm   1/2     Running    109 (2m27s ago)   9h
rook-ceph-tools-ff7b6fb65-gpndb                    1/1     Running    0                 9h
```

**Comment by vyom-soft** [comment]
> > Hi. Not able to open the `rook-ceph-operator.txt` file that you have attached in the above comment. Can you please provide the rook operator pod logs. That would help in debugging.
> 
> Hi, Thank you for you time. The fresh operator log. Please see the same. [rook-ceph-operator-log.txt](https://github.com/rook/rook/files/11118628/rook-ceph-operator-log.txt) Any other logs or information is required.?

Fresh operator log. 
[operator-log.txt](https://github.com/rook/rook/files/11119046/operator-log.txt)

**Comment by sp98** [comment]
Thanks for the logs.  Looks like the issue is around creating service monitors for mgr pods. So its not even reaching the point where OSDs creation starts:

```
2023-03-31 05:58:33.565242 D | ceph-spec: CephCluster "rook-ceph" status: "Progressing". "failed to create cluster: failed to start ceph mgr: failed to enable mgr services: failed to enable service monitor: service monitor could not be enabled: failed to create servicemonitor. the server could not find the requested resource (post servicemonitors.monitoring.coreos.com)"
2023-03-31 05:58:33.585908 D | ceph-spec: found 1 ceph clusters in namespace "rook-ceph"
2023-03-31 05:58:33.585936 D | ceph-cluster-controller: update event on CephCluster CR
2023-03-31 05:58:33.587485 E | ceph-cluster-controller: failed to reconcile CephCluster "rook-ceph/rook-ceph". failed to reconcile cluster "rook-ceph": failed to configure local ceph cluster: failed to create cluster: failed to start ceph mgr: failed to enable mgr service
```

**Comment by sp98** [comment]
From the `values.yaml` file you have shared :

``` 
# monitoring requires Prometheus to be pre-installed
monitoring:
  # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
  enabled: true```

Probably you don't have prometheus preinstalled. Can you please check that. Thanks.

**Comment by sp98** [comment]
> > From the `values.yaml` file you have shared :
> > ```
> > # monitoring requires Prometheus to be pre-installed
> > monitoring:
> >   # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
> >   enabled: true
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Probably you don't have prometheus preinstalled. Can you please check that. Thanks.
> 
> Yes, Thank you for pointing it out. You are correct. I was also inspecting the same entry. I do not have Prometheus installed as of now. It is brand new k8s cluster setup, with new devices. I will verify on that (probably I will install prometheus) & would expect Chep to pick it up automatically?.

Yes.

**Comment by vyom-soft** [comment]
> > > From the `values.yaml` file you have shared :
> > > ```
> > > # monitoring requires Prometheus to be pre-installed
> > > monitoring:
> > >   # enabling will also create RBAC rules to allow Operator to create ServiceMonitors
> > >   enabled: true
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Probably you don't have prometheus preinstalled. Can you please check that. Thanks.
> > 
> > 
> > Yes, Thank you for pointing it out. You are correct. I was also inspecting the same entry. I do not have Prometheus installed as of now. It is brand new k8s cluster setup, with new devices. I will verify on that (probably I will install prometheus) & would expect Chep to pick it up automatically?.
> 
> Yes.

Thank you. OSD are up and running after I installed Prometheus in the K8s Custer

<img width="476" alt="image" src="https://user-images.githubusercontent.com/18686082/229300154-682f5a40-ca7b-46d8-b8b0-6494c8a4fb3c.png">

<img width="664" alt="image" src="https://user-images.githubusercontent.com/18686082/229300437-af3cb4c1-fb7f-4f83-b91b-454d0191cfa5.png">

**Comment by ZYWNB666** [comment]
I had the same problem as my friend here

```
[rook@rook-ceph-tools-5d4db8cfc5-9l2ww /]$ ceph status
  cluster:
    id:     f6527471-0bda-40af-b97b-ff29d6ad3fff
    health: HEALTH_WARN
            Reduced data availability: 1 pg inactive
            OSD count 0 < osd_pool_default_size 3

  services:
    mon: 3 daemons, quorum a,b,c (age 24m)
    mgr: b(active, since 21m), standbys: a
    osd: 0 osds: 0 up, 0 in

  data:
    pools:   1 pools, 1 pgs
    objects: 0 objects, 0 B
    usage:   0 B used, 0 B / 0 B avail
    pgs:     100.000% pgs unknown
             1 unknown

[rook@rook-ceph-tools-5d4db8cfc5-9l2ww /]$ ceph osd tree
ID  CLASS  WEIGHT  TYPE NAME     STATUS  REWEIGHT  PRI-AFF
-1              0  root default
[rook@rook-ceph-tools-5d4db8cfc5-9l2ww /]$ ceph df
--- RAW STORAGE ---
CLASS  SIZE  AVAIL  USED  RAW USED  %RAW USED
TOTAL   0 B    0 B   0 B       0 B          0

--- POOLS ---
POOL            ID  PGS  STORED  OBJECTS  USED  %USED  MAX AVAIL
ceph-blockpool   1    1     0 B        0   0 B      0        0 B
[rook@rook-ceph-tools-5d4db8cfc5-9l2ww /]$
```


There are also such errors in the rook-ceph-operator pod

```
2024-08-06 10:59:44.796525 I | clusterdisruption-controller: all "host" failure domains: []. osd is down in failure domain: "". active node drains: false. pg health: "cluster is not fully clean. PGs: [{StateName:unknown Count:1}]"
2024-08-06 10:59:44.807935 I | clusterdisruption-controller: reconciling osd pdb reconciler as the allowed disruptions in default pdb is 0
```
[rook-ceph-mgr-b-7549d6f4b8.log](https://github.com/user-attachments/files/16509304/rook-ceph-mgr-b-7549d6f4b8.log)


These are mgr logs

**Comment by sp98** [comment]
@ZYWNB666 
``` osd: 0 osds: 0 up, 0 in
```
means none of the OSD daemons are running. So need to check why the OSDs are not running. 

Can you share: 
- `kubectl get pods -n rook-ceph` 
- rook-ceph-operator pod logs
- Logs of the rook-ceph-osd-prepare pods.

**Comment by ZYWNB666** [comment]
> @ZYWNB666
> 
> means none of the OSD daemons are running. So need to check why the OSDs are not running.
> 
> Can you share:
> 
> * `kubectl get pods -n rook-ceph`
> * rook-ceph-operator pod logs
> * Logs of the rook-ceph-osd-prepare pods.
> * rook-ceph cluster yaml

Thanks for your reply, I have found the cause of the problem, because the disk was not cleaned after using ceph last time, I cleaned it again, and now it is working normally

**Comment by phekno** [comment]
I had/have a similar issue.  Blew away rook-ceph, re-installed it, the osd-prepare pods wouldn't come up.  Came across this same error and found this issue.

All that said, even WITH Prometheus installed, I couldn't get rook to progress to the point where it would provision the OSDs.  Not sure if there's something missing with my Prometheus install or what...

**Comment by travisn** [comment]
@phekno Check if you cleaned up the hosts from the previous install according to the [Cleanup guide](https://rook.io/docs/rook/latest/Getting-Started/ceph-teardown/#delete-the-data-on-hosts). The new cluster typically fails to start in a re-install due to not fully cleaning up the previous install.

**Comment by phekno** [comment]
> @phekno Check if you cleaned up the hosts from the previous install according to the [Cleanup guide](https://rook.io/docs/rook/latest/Getting-Started/ceph-teardown/#delete-the-data-on-hosts). The new cluster typically fails to start in a re-install due to not fully cleaning up the previous install.

Yep...I had cleaned out the hosts.  About the only thing I had a hiccup with (and maybe this is the problem) is that one of the ConfigMaps and a Secret couldn't be deleted.  Their finalizers were still set.  When I tried to un-set those finalizers, that didn't work, either.  I did manage to destroy the namespace, but those resources were orphaned.  I guess I'm not 110% sure if that is/was the case or not.  I still might try the entire cleanup process again to see if it was those orphaned resources...
